app/src/main/java/com/example/anfcodetest/CustomViewHolder.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.anfcodetest
+
+import android.content.Intent
+import android.text.Html
+import android.text.method.LinkMovementMethod
+import android.view.View
+import android.widget.Button
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.card_row.view.*
+
+class CustomViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
```

Comment:
So here I implemented the view holder, as this should be what holds and sets all the data and views on your RecyclerView. I took the card logic from your presenter, and did all the visibility work here! This adds a few things to our implementation:
- It allows our view holder to clearly own the views that it already holds in XML
- It is much easier to test
- We can easily pass in data via the `bind` method and don't have to iterate over it in our presenter
- It allows our MainActivity to own our RecyclerView